src/language/tokenKind.ts

```diff
@@ -2,29 +2,29 @@
  * An exported enum describing the different kinds of tokens that the
  * lexer emits.
  */
-enum TokenKind {
-  SOF = '<SOF>',
-  EOF = '<EOF>',
-  BANG = '!',
-  DOLLAR = '$',
-  AMP = '&',
-  PAREN_L = '(',
-  PAREN_R = ')',
-  SPREAD = '...',
-  COLON = ':',
-  EQUALS = '=',
-  AT = '@',
-  BRACKET_L = '[',
-  BRACKET_R = ']',
-  BRACE_L = '{',
-  PIPE = '|',
-  BRACE_R = '}',
-  NAME = 'Name',
-  INT = 'Int',
-  FLOAT = 'Float',
-  STRING = 'String',
-  BLOCK_STRING = 'BlockString',
-  COMMENT = 'Comment',
-}
-
-export { TokenKind };
+export const TokenKind = {
+  SOF: '<SOF>' as const,
+  EOF: '<EOF>' as const,
+  BANG: '!' as const,
+  DOLLAR: '$' as const,
+  AMP: '&' as const,
+  PAREN_L: '(' as const,
+  PAREN_R: ')' as const,
+  SPREAD: '...' as const,
+  COLON: ':' as const,
+  EQUALS: '=' as const,
+  AT: '@' as const,
+  BRACKET_L: '[' as const,
+  BRACKET_R: ']' as const,
+  BRACE_L: '{' as const,
+  PIPE: '|' as const,
+  BRACE_R: '}' as const,
+  NAME: 'Name' as const,
+  INT: 'Int' as const,
+  FLOAT: 'Float' as const,
+  STRING: 'String' as const,
+  BLOCK_STRING: 'BlockString' as const,
+  COMMENT: 'Comment' as const,
+} as const;
+// eslint-disable-next-line @typescript-eslint/no-redeclare
+export type TokenKind = (typeof TokenKind)[keyof typeof TokenKind];
```

Comment:
I think it might be a bit confusing to have the type and the raw value be the same,

if somebody in their code wants to import both in the same file, I think the identifiers clash.

same for the others.

@benjie had a suggestion: https://github.com/graphql/graphql-js/pull/3360#issuecomment-963053990

Reply:
x however many "enums" we have

Reply:
That's an opinion though 😅 I personally think this is easier to read

Reply:
is it? i am getting an error in my editor when i have the following:

```ts
import type { A } from 'somewhere.js' 
import { A } from 'somewhere.js' 
```

although there is no error with the following:

```ts
export type { A } from 'somewhere.js' 
export { A } from 'somewhere.js' 
```

for our users, they will be using the former case.

I could be wrong here, I don't think I've encountered this before.

Reply:
Because you only need `import { A } from 'somewhere.js' ` as that is both a type as well as a value.

Reply:
Anyway my preference for this approach was 'least amount of breaking changes', if we can't agree on this approach then I welcome any other one.

Reply:
> Because you only need `import { A } from 'somewhere.js' ` as that is both a type as well as a value.

Ah! I see, that should be fine then.

Reply:
@jasonkuhrt thanks for joining in, I am following along also with interest at #4253 and #4268